pkg/monitor/cluster/cluster.go

```diff
@@ -130,18 +130,18 @@ func (mon *Monitor) Monitor(ctx context.Context) (errs []error) {
 		})
 	}
 
-	// If API is not returning 200, don't need to run the next checks
-	statusCode, err := mon.emitAPIServerHealthzCode(ctx)
-	if err != nil {
+	// If API Server is unreachable, don't need to run the next checks
+	statusCode, err := mon.getAPIServerPingCode(ctx)
+	if err != nil || statusCode != http.StatusOK {
 		errs = append(errs, err)
-		friendlyFuncName := steps.FriendlyName(mon.emitAPIServerHealthzCode)
+		friendlyFuncName := steps.FriendlyName(mon.getAPIServerPingCode)
```

Comment:
What's the advantage of using this metric (/healthz/ping) over what we use now (just /healthz) for the initial check?

The new metric apiserver.healthz.ping.code makes sense to me, just not sure why we would replace the existing metric here.

Reply:
We're enabling [Azure Resource Health for ARO](https://learn.microsoft.com/en-us/azure/service-health/resource-health-overview). This requires us to emit metrics (aka "signal") to a Geneva monitor, which alerts Azure Resource Health.

According to the PRD, customers want a signal for cluster reachability. One of the reachability scenarios is API Server reachability, hence this PR. APIServerHealthz code incorporates many tests, and the best one to represent reachability is the ping test.